### ISSUE TITLE
Python 3.11 Installation Error (Missing 3.11 Binary/Wheel)

### ISSUE THREAD

**Issue by Warfront1** [pymssql/pymssql#766]
When attempting to upgrade from Python 3.10 to Python 3.11, the installation process does not work the same for pymssql.
Given the same exact environment: Python 3.11 will **fail** to install; where as 3.10 has no issues.

For complete clarity, I have scripted out the environment into the following `Dockerfile`
```
FROM ubuntu:22.04

RUN apt-get update && apt install software-properties-common -y && add-apt-repository ppa:deadsnakes/ppa -y && \
     apt update
ARG DEBIAN_FRONTEND=noninteractive
ARG PYTHON_VERSION
RUN apt install -y python$PYTHON_VERSION python$PYTHON_VERSION-dev python$PYTHON_VERSION-distutils python3-pip && \
    python$PYTHON_VERSION -m pip install --upgrade pip
RUN python$PYTHON_VERSION -m pip install pymssql==2.2.5
```

The following build command runs flawlessly:
`docker build -t pymssql-testbed . --build-arg PYTHON_VERSION=3.10`

The following build command fails:
`docker build -t pymssql-testbed . --build-arg PYTHON_VERSION=3.11`

<details><summary>Error Message Received:</summary>
<p>

```
#7 0.615 Collecting pymssql==2.2.5
#7 0.739   Downloading pymssql-2.2.5.tar.gz (170 kB)
#7 0.789      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 170.1/170.1 kB 3.5 MB/s eta 0:00:00
#7 0.922   Installing build dependencies: started
#7 4.683   Installing build dependencies: finished with status 'done'
#7 4.691   Getting requirements to build wheel: started
#7 4.975   Getting requirements to build wheel: finished with status 'done'
#7 4.982   Preparing metadata (pyproject.toml): started
#7 5.196   Preparing metadata (pyproject.toml): finished with status 'done'
#7 5.198 Building wheels for collected packages: pymssql
#7 5.198   Building wheel for pymssql (pyproject.toml): started
#7 6.497   Building wheel for pymssql (pyproject.toml): finished with status 'error'
#7 6.500   error: subprocess-exited-with-error
#7 6.500
#7 6.500   × Building wheel for pymssql (pyproject.toml) did not run successfully.
#7 6.500   │ exit code: 1
#7 6.500   ╰─> [22 lines of output]
#7 6.500       setup.py: platform.system() => Linux
#7 6.500       setup.py: platform.architecture() => ('64bit', 'ELF')
#7 6.500       setup.py: platform.libc_ver() => ('glibc', '2.35')
#7 6.500       setup.py: include_dirs => []
#7 6.500       setup.py: library_dirs => []
#7 6.500       running bdist_wheel
#7 6.500       running build
#7 6.500       running build_py
#7 6.500       creating build
#7 6.500       creating build/lib.linux-x86_64-3.11
#7 6.500       creating build/lib.linux-x86_64-3.11/pymssql
#7 6.500       copying src/pymssql/__init__.py -> build/lib.linux-x86_64-3.11/pymssql
#7 6.500       running build_ext
#7 6.500       creating build/temp.linux-x86_64-3.11
#7 6.500       creating build/temp.linux-x86_64-3.11/src
#7 6.500       creating build/temp.linux-x86_64-3.11/src/pymssql
#7 6.500       x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.11 -c src/pymssql/_mssql.c -o build/temp.linux-x86_64-3.11/src/pymssql/_mssql.o 
-DMSDBLIB
#7 6.500       src/pymssql/_mssql.c:699:10: fatal error: sqlfront.h: No such file or directory
#7 6.500         699 | #include "sqlfront.h"
#7 6.500             |          ^~~~~~~~~~~~
#7 6.500       compilation terminated.
#7 6.500       error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
#7 6.500       [end of output]
```

</p>
</details>

**Comment by Warfront1** [comment]
@tirkarthi It would appear to be related, but not the same.
I believe the root cause to be related to the binary for 3.11 being missing.
Observe the following builds when attempting to install via pip with the flag [--only-binary :all:](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-only-binary)
| Python Version | Has Binary |
| ------------- | ------------- |
| 3.11 | FALSE |
| 3.10 | TRUE |
| 3.9 | TRUE |
| 3.8 | TRUE |
| 3.7 | TRUE |


**For those who need a workaround until this gets resolved:**
1. Downgrade pip to `21.2.4` or lower
2. Install `freetds-dev` and  `libssl-dev`

<details><summary>Complete Docker File with 3.11 workaround:</summary>
<p>

```
FROM ubuntu:22.04

RUN apt-get update && apt install software-properties-common -y && add-apt-repository ppa:deadsnakes/ppa -y && \
     apt update
ARG DEBIAN_FRONTEND=noninteractive
ARG PYTHON_VERSION
RUN apt install -y \
    python$PYTHON_VERSION python$PYTHON_VERSION-dev python$PYTHON_VERSION-distutils python3-pip && \
    python$PYTHON_VERSION -m pip install -I pip==21.2.4
RUN apt install -y freetds-dev libssl-dev
RUN python$PYTHON_VERSION -m pip install pymssql==2.2.5
```

</p>
</details>

**Comment by invokermain** [comment]
Yep the library really needs to provide wheels for 3.11. This is definitely one of the 'harder to compile' packages.

**Comment by rliessum** [comment]
On MacOS (12.6) I did as follows (I use fish shell, for other shells, export it):
- Downgrade to PIP 21.2.4
- brew unlink openssl && brew link openssl --force
- set -gx LDFLAGS "-L/usr/local/opt/openssl@3/lib"
- set -gx CPPFLAGS "-I/usr/local/opt/openssl@3/include"
- set -gx PKG_CONFIG_PATH "/usr/local/opt/openssl@3/lib/pkgconfig"
- fish_add_path /usr/local/opt/openssl@3/bin
- pip install pymssql

And I was good to go.

**Comment by brianoflondon** [comment]
This was fixed for me on Ubuntu 22.04 LTS with one line from the other open Issue:

```terminal
sudo apt install freetds-dev
```

I believe I have `libssl-dev` already.

**Comment by invokermain** [comment]
yep good luck getting this compiled on windows though, it's very unclear as to what provides the require sql header files for the compilation. I just downgraded back to 3.10.

@termim any plans to release 3.11 wheels?

**Comment by zschumacher** [comment]
also interested in this - blocking adding support for 3.11 to pydapper (zschumacher/pydapper#84)

**Comment by siggemannen** [comment]
I hate to write "me too", but i'm also interested in a windows python 3.11 version of pymssql. Getting hit with missing header error as well

**Comment by termim** [comment]
Python 3.11 wheels are published for pymssql-2.2.6.

**Comment by irufus** [comment]
@termim what about for pymssql-2.2.7?
